ibis/expr/deferred.py

```diff
@@ -20,6 +20,11 @@ def __init__(self, resolve: Callable = toolz.identity) -> None:
         ), f"resolve argument is not callable, got {type(resolve)}"
         self.resolve = resolve
 
+    def __hash__(self) -> int:
```

Comment:
I guess this is the default implementation by python, need to verify though.

Reply:
It's only the default if you leave `__eq__` and `__hash__` **both** not implemented:


```
In [1]: class Foo:
   ...:     def __eq__(self, other):
   ...:         return True
   ...:

In [2]: hash(Foo())
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In [2], line 1
----> 1 hash(Foo())

TypeError: unhashable type: 'Foo'
```

Reply:
We override `__eq__` to have the expression behavior, so we have to override `__hash__` to get any kind of hashing at al.